### DISCUSSION TITLE
Training Model Head/Neck CTs with groundtruth segmentation and volume .nrrd files

### DISCUSSION THREAD

**Discussion by danielsh3n** [Project-MONAI/MONAILabel#1593] [Q&A]
I am hoping to train a 3D U-Net model on lymphatic/fatty tissue in the head and neck. Regarding my dataset, I have 18 cases along with their manual segmentations (done in 3DSlicer) in .nrrd files. Is it possible for me to use MONAILabel to train and test on this dataset?

@hw-ju

**Comment by diazandr3s** [comment]
Hi @danielsh3n,

This is a good question. Certainly, you can train a MONAI Label model for this. 

Please follow [these instructions](https://youtu.be/KtPE8m0LvcQ?si=vylb4tNB4Rfnv81a&t=333)

They are simple.

Once you create the Python env, install monai label and download the Radiology app (https://github.com/Project-MONAI/MONAILabel/tree/main#getting-started-with-monai-label).

- Update the label names [here](https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation.py#L37-L62) with the consistent label indexes for the segment you already have
- As training from scratch, disable or set to false this flag: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation.py#L72
- Organize the dataset in the MONAI Label structure
- Train

 let us know how that goes

**Reply by danielsh3n** [reply]
Hi @diazandr3s ,

Thanks for getting back to me! I have the radiology app downloaded, updated my label names (just 'CTV':1) for the target volume I care about, disabled the flag, and have my dataset organized like so:

C:.
│   case_1.nrrd
│   case_10.nrrd
│   case_11.nrrd
│   case_12.nrrd
│   case_13.nrrd
│   case_14.nrrd
│   case_15.nrrd
│   case_16.nrrd
│   case_17.nrrd
│   case_18.nrrd
│   case_2.nrrd
│   case_3.nrrd
│   case_4.nrrd
│   case_5.nrrd
│   case_6.nrrd
│   case_7.nrrd
│   case_8.nrrd
│   case_9.nrrd
│
└───labels
    └───final
            case_1.nrrd
            case_10.nrrd
            case_11.nrrd
            case_12.nrrd
            case_13.nrrd
            case_14.nrrd
            case_15.nrrd
            case_16.nrrd
            case_17.nrrd
            case_18.nrrd
            case_2.nrrd
            case_3.nrrd
            case_4.nrrd
            case_5.nrrd
            case_6.nrrd
            case_7.nrrd
            case_8.nrrd
            case_9.nrrd
            
Can you elaborate on how exactly I go about training the model from these labels? Do I open each volume .nrrd file along with the corresponding segmentation .nrrd file in Slicer and then press train? Any resources on this would be greatly appreciated!

Thanks,
Daniel

**Reply by diazandr3s** [reply]
Hi @danielsh3n,

It is great! You already have most of the things done.

To trigger the training, please follow these instructions: https://youtu.be/3HTh2dqZqew?si=fQVJ_9fSuMgpoi2w&t=194

As you already have the labels, you don't need to submit anything in Slicer. Just connect to the server and trigger the training with the default configuration.

Let us know how that goes,

**Reply by danielsh3n** [reply]
Hi @diazandr3s ,

Thanks for the quick response! Glad to hear that I'm almost there! Unfortunately, when I tried starting the server with the new path to my dataset, I get these errors:

![MONAI-Label_issue](https://github.com/Project-MONAI/MONAILabel/assets/77021358/3b56b330-5842-4f70-a00f-2e71935814ff)

**Reply by diazandr3s** [reply]
Hi @danielsh3n,

It seems an error from the Python env. Can you please create a fresh Python environment and try again?
BTW, no need to use double quotes for the _studies_ argument.

Let us know,

**Reply by danielsh3n** [reply]
Hi @diazandr3s ,

I just set up with a new environment and connected to the server in Slicer and have these outputs into Slicer and in terminal (WSL2 on Ubuntu 22.04). Thoughts? Also, after training the model, how would I go about testing the model's predictions on a new head/neck CT? 

![image](https://github.com/Project-MONAI/MONAILabel/assets/77021358/cd8bd15b-dd1e-4e36-bd6f-58b8286a8182)

[(monailabel) dshen101@LAPTOP-PMB3NU.txt](https://github.com/Project-MONAI/MONAILabel/files/13573169/monailabel.dshen101%40LAPTOP-PMB3NU.txt)


Thanks!

**Reply by diazandr3s** [reply]
Hi @danielsh3n,

Sorry for my late reponse.
I checked the logs and it seems there is an issue with the default transforms or the labels. 
Are there any empty labels? Did you make sure the label (CTV) is represented by 1 in all the label volumes?
Do you know the Hounsfield units range in which these regions are more visible?

Let us know,
